File: ChatClient/ChatClient/Logic/Global.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChatClient
{
    public class Global
    {
        #region 网络相关
        public static bool SendData(TCPOutPacket tcpOutPacket)
        {
            return Program.tcpClient.SendData(tcpOutPacket);
        }
        public static bool SendData(int nID, int cmd, byte[] data)
        {
            return Program.tcpClient.SendData(TCPOutPacket.MakeTCPOutPacket(nID, cmd, data));
        }
        #endregion

        #region 日志
        private static string _ExceptionPath = AppDomain.CurrentDomain.BaseDirectory + @"Exception/";
        public static void WriteExceptionEx(string msg, Exception ex = null)
        {
            string exMsg = "";
            if (ex != null)
            {
                exMsg = ex.ToString();
            }
            try
            {
                if (!System.IO.Directory.Exists(_ExceptionPath))
                {
                    System.IO.Directory.CreateDirectory(_ExceptionPath);
                }
                StreamWriter sw = File.CreateText(_ExceptionPath + DateTime.Now.ToString("yyyyMMddHHmmssffff") + ".log");
                sw.WriteLine(msg + "\r\n" + exMsg);
                sw.Close();
                sw = null;
            }
            catch
            {

            }
        }
        #endregion
    }
}

File: chat_server/logic/global.cpp
#include"global.h"

DEFINE_SINGLETON(Global);


TcpOutPacket* Global::MakeTCPOutPacket(int nID, int cmd, string data)
{
	return new TcpOutPacket(nID,cmd,data.c_str());
}

bool Global::SendData(int fd, TcpOutPacket* tcpOutPacket)
{
	if(send(fd, tcpOutPacket->getPacketBytes(), tcpOutPacket->_packet_size, 0)<0)
	{
		err_log("SendData Error");
		return false;
	}
	return true;
}

bool Global::SendData(int fd, int nID, int cmd, string data)
{
	return SendData(fd, MakeTCPOutPacket(nID, cmd, data));
}

